src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1350,6 +1355,11 @@ private static Type GetMemberClrType(MemberInfo memberInfo)
                                     isExtensionData = true;
                                 }
                                 break;
+                            case JsonRequiredAttributeFullName:
+                                {
+                                    isRequired = true;
```

Comment:
Any reason to not support the `required` keyword for src-gen yet? Do we also need to check for `RequiredMemberAttribute`? If not, are we gonna issue a diagnostic per https://github.com/dotnet/runtime/issues/70945?

Reply:
It would require reworking the `JsonParameterInfo` infrastructure to also incorporate `required`/`init` properties so we've punted it for now. @krwq given that #70945 have now been closed could you open an issue tracking this work. You might consider closing #58770 and fusing both into one user story.

Reply:
I've created https://github.com/dotnet/runtime/issues/73219 to track this

Reply:
Can we close #58770 as well to minimize duplication?